### DISCUSSION TITLE
Laragon auto deleting php.ini

### DISCUSSION THREAD

**Discussion by JawadR1** [leokhoa/laragon#531] [Q&A] [answered]
Hi,
I have installed Laragon 6.0.0 which came with `php-8.1.10-Win32-vs16-x64`. I also want to add `php-7.4.33-Win32-vc15-x64` to the installation. I added PHP files in `C:\laragon\bin\php` and created a `php.ini` file. But when I change the PHP version and go to reload the Laragon/Apache, it deletes the `php.ini` file from PHP7.4 folder but the file in PHP8.1 remain there. I have even tried installing the Laragon again but same issue. Please let me know how can I stop it from deleting the file.

Thanks.

_**Edit 1:**_ Also, I should note that I was using 4 PHP versions with Laragon with before my last OS reinstall and everything was working fine. All of the PHP versions had their own `php.ini` files too.

**Comment by Pen-y-Fan** [comment]
Usually, there is no need to create the php.ini manually, Laragon does this automatically. See my blog [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/) for step-by-step instructions.

**Reply by JawadR1** [reply]
I have read your blog but that didn't help me fix the issue.

I think it does need `php.ini` (in my case at least)

**Missing arrow for extensions**
![Missing arrow for extensions](https://github.com/leokhoa/laragon/assets/34421367/f0ca45f0-2876-4465-82fb-674dd71c25db)

**When I click on `php.ini` in the above image**
![error](https://github.com/leokhoa/laragon/assets/34421367/13975b5b-a5a7-448c-ad51-50f8d6d318b4)

**Extensions with both versions**
![extenstions](https://github.com/leokhoa/laragon/assets/34421367/146fb365-ad21-4b32-8673-ccc3795aee96)

How can I add extensions to `v7.4` if I can't find `php.ini` file

**Reply by Pen-y-Fan** [reply, marked as answer]
Looking at the screenshots Laragon has picked up the PHP 7.4 in the PHP directory. The next thing I can think of is the directory is corrupt. Can you:

- delete the **php-7.4.33-Win32-vc15-x64** directory from laragon
- download **php-7.4.32-Win32-vc15-x64.zip** from <https://windows.php.net/downloads/releases/archives/>
    - oddly there isn't a 7.4.33 in the archive, 7.4.32 will have the same functionality.
- follow the blog, unzip in Laragon's php directory. etc. etc.

Please let us know how you get on.

**Reply by JawadR1** [reply]
You are right. The directory was corrupted for some reason. Redownloading the PHP files fixed the issue.
I actually tried adding 3 different PHP versions (7.4, 8.0, 8.2), and all of those had the same problem so I didn't consider redownloading the PHP files.

Thanks for the help.
